### PR TITLE
Nf/added confirm modal for revoking key serials

### PR DESCRIPTION
**Pull request by NabilFurmoli**

### CODE REVIEW

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            return;
+        }
+        this.setState({ submitting: true });
+        await this.props.onRevoke(this.props.selectedKeySerial);
+        this.setState({ submitting: false });
+        this.props.closeModal();
+    };
+
+    private _isValidToRevoke = () => {
+        return this.props.selectedKeySerial !== null;
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: this should also check if the key serial has an assignment

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        this.props.updateSelectedKeySerial(keySerial);
+    };
+
+     private _revokeEquipment = async () => {
+        if (!this._isValidToRevoke()) {
+            return;
+        }
+        this.setState({ submitting: true });
+        await this.props.onRevoke(this.props.selectedKeySerial);
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: i added this to the other revokes in my PR, but you might as well add try/catching in this one. 
```
this.setState({ submitting: true });
        try {
            await this.props.onRevoke(this.props.selectedKeySerial);
        } catch (err) {
            this.setState({ submitting: false });
            return;
        }
        this.setState({ submitting: false });
        this.props.closeModal();
```

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    public context: AppContext;
+    constructor(props) {
+        super(props);
+        this.state = {
+            submitting: false
+        };
+    }
+
+    public componentDidMount() {
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: we aren't fetching the data on the other revokes, but i'd wait to remove this until we get @srkirkland's opinion

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                            Revoke{" "}
+                            {this.state.submitting && <i className="fas fa-circle-notch fa-spin" />}
+                        </Button>{" "}
+                    </ModalFooter>
+                </Modal>
+            </div>
+        );
+    }
+
+    private _revokeEquipment = async () => {
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: @NabilFurmoli missed this the first time, but `_revokeEquipment` -> `_revokeKeySerial`

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    onRevoke: (keySerial: IKeySerial) => void;
+    selectedKeySerial: IKeySerial;
+    updateSelectedKeySerial: (keySerial: IKeySerial) => void;
+}
+
+interface IState {
+    submitting: boolean;
+}
+
+export default class RevokeKeySerials extends React.Component<IProps, IState> {
```
Comment (laholstege) [inline Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: pretty minor but `RevokeKeySerials` -> `RevokeKeySerial`

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx (hunk 4 above)
Comment (NabilFurmoli) [reply Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: fixed

File: Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx (hunk 5 above)
Comment (NabilFurmoli) [reply Keas.Mvc/ClientApp/components/Keys/RevokeKeySerials.tsx]: fixed

**Pull request merged**